src/agent/src/agent.cpp

```diff
@@ -35,6 +35,7 @@ Agent::Agent(const std::string& configFilePath, std::unique_ptr<ISignalHandler>
     // Check if agent is registered
     if (m_agentInfo.GetName().empty() || m_agentInfo.GetKey().empty() || m_agentInfo.GetUUID().empty())
     {
+        LogCritical("The agent is not registered");
```

Comment:
That message was added on purpose because if the agent is not registered it crashes without any clue. This is due because the exception is not being catch anywhere.

Reply:
If I'm not mistaken this has already been addressed and merged to `master`.

Reply:
Yes, I've just rebased and I'm testing precisely this to ensure only the exception message is logged. If successful, I will remove the line.